go/store/types/map.go

```diff
@@ -399,26 +399,28 @@ func (m Map) IteratorBackFrom(ctx context.Context, key Value) (MapIterator, erro
 		cur.advance(ctx)
 	}
 
-	item, err := cur.current()
-
-	if err != nil {
-		return nil, err
-	}
+	if cur.valid() {
+		item, err := cur.current()
 
-	entry := item.(mapEntry)
-	isLess, err := entry.key.Less(m.Format(), key)
-
-	if err != nil {
-		return nil, err
-	}
+		if err != nil {
+			return nil, err
+		}
 
-	if !isLess && !key.Equals(entry.key) {
-		_, err := cur.advance(ctx)
+		entry := item.(mapEntry)
+		isLess, err := entry.key.Less(m.Format(), key)
 
 		if err != nil {
 			return nil, err
 		}
-	}
+
+		if !isLess && !key.Equals(entry.key) {
+			_, err := cur.advance(ctx)
+
+			if err != nil {
+				return nil, err
+			}
+		}
+	} // else: we're at the beginning of the map
```

Comment:
IDK what's happening with this diff. I added lines 402 and 423

Reply:
Change looks good.  Not sure what you are asking.

If the map has at least one element in it the 3 cases (key < map element key, key == map element key, key > map element key) will all return a valid MapIterator even if it has no valid elements.

In the case of an empty map it sees it's passed the last element, goes back one and falsely assumes there is a valid element there... but there isn't.